dali/c_api_2/pipeline.cc

```diff
@@ -0,0 +1,302 @@
+#include "dali/c_api_2/utils.h"
+#include "dali/c_api_2/validation.h"
+
+namespace dali::c_api {
+
+PipelineWrapper *ToPointer(daliPipeline_h handle) {
+  if (!handle)
+    throw NullHandle("Pipeline");
+  return static_cast<PipelineWrapper *>(handle);
+}
+
+PipelineWrapper::PipelineWrapper(const daliPipelineParams_t &params) {
+  bool pipelined = true, async = true, dynamic = false, set_affinity = false;
+  if (params.exec_type_present) {
+    pipelined = params.exec_type & DALI_EXEC_IS_PIPELINED;
+    async = params.exec_type & DALI_EXEC_IS_ASYNC;
+    dynamic = params.exec_type & DALI_EXEC_IS_DYNAMIC;
+  }
+  if (params.exec_flags_present) {
+    set_affinity = params.exec_flags & DALI_EXEC_FLAGS_SET_AFFINITY;
+  }
+  pipeline_ = std::make_unique<Pipeline>(
+    GET_OR_DEFAULT(params, max_batch_size, -1),
+    GET_OR_DEFAULT(params, num_threads, -1),
+    GET_OR_DEFAULT(params, device_id, -1),
+    GET_OR_DEFAULT(params, seed, -1_i64),
+    pipelined,
+    GET_OR_DEFAULT(params, prefetch_queue_depth, 2),
+    async,
+    dynamic,
+    0,
+    set_affinity);
+}
+
+PipelineWrapper::PipelineWrapper(
+      const void *serialized,
+      size_t length,
+      const daliPipelineParams_t &params) {
+  bool pipelined = true, async = true, dynamic = false, set_affinity = false;
+  if (params.exec_type_present) {
+    pipelined = params.exec_type & DALI_EXEC_IS_PIPELINED;
+    async = params.exec_type & DALI_EXEC_IS_ASYNC;
+    dynamic = params.exec_type & DALI_EXEC_IS_DYNAMIC;
+  }
+  if (params.exec_flags_present) {
+    set_affinity = params.exec_flags & DALI_EXEC_FLAGS_SET_AFFINITY;
+  }
+  pipeline_ = std::make_unique<Pipeline>(
+    std::string(static_cast<const char *>(serialized), length),
+    GET_OR_DEFAULT(params, max_batch_size, -1),
+    GET_OR_DEFAULT(params, num_threads, -1),
+    GET_OR_DEFAULT(params, device_id, -1),
+    pipelined,
+    GET_OR_DEFAULT(params, prefetch_queue_depth, 2),
+    async,
+    dynamic,
+    0,
+    set_affinity,
+    GET_OR_DEFAULT(params, seed, -1_i64));
+}
+
+PipelineWrapper::~PipelineWrapper() = default;
+
+std::unique_ptr<PipelineOutputs>
+PipelineWrapper::PopOutputs(AccessOrder order) {
+  return std::make_unique<PipelineOutputs>(pipeline_.get(), order);
+}
+
+void PipelineWrapper::Build() {
+  pipeline_->Build();
+}
+
+void PipelineWrapper::Run() {
+  pipeline_->Run();
+}
+
+void PipelineWrapper::Prefetch() {
+  pipeline_->Prefetch();
+}
+
+int PipelineWrapper::GetFeedCount(std::string_view input_name) {
+  return pipeline_->InputFeedCount(input_name);
+}
+
+void PipelineWrapper::FeedInput(
+    std::string_view input_name,
+    const ITensorList *input_data,
+    std::optional<std::string_view> data_id,
+    daliFeedInputFlags_t options,
+    std::optional<cudaStream_t> stream) {
+  assert(input_data);
+  if (input_data->GetBufferPlacement().device_type == DALI_STORAGE_CPU) {
+    FeedInputImpl(
+      input_name,
+      *input_data->Unwrap<CPUBackend>(),
+      data_id,
+      options,
+      stream);
```

Comment:
I think those fit in line length limit, at least that happened when I clang-formatted that code.

Reply:
It looks clear written like this, I will keep it.